### PR TITLE
[FEAT] payprov2

### PR DESCRIPTION
**Pull request by gabrielbazan7**

### CODE REVIEW

File: packages/bitcore-wallet-client/test/utils.test.js [hunk 1]
```diff
@@ -3,8 +3,8 @@
 var _ = require('lodash');
 var chai = require('chai');
 var should = chai.should();
-var Bitcore = require('crypto-wallet-core').BitcoreLib;
-var { Utils } = require('../ts_build/common');
+var Bitcore = require('bitcore-lib');
```
Comment (justinkook) [inline packages/bitcore-wallet-client/test/utils.test.js]: revert back to using CWC.BitcoreLib

File: packages/bitcore-wallet-client/src/lib/payproV2.ts [hunk 2]
```diff
@@ -0,0 +1,429 @@
+'use strict';
+
+// Native
+const superagent = require('superagent');
+const query = require('querystring');
+const url = require('url');
+const dfltTrustedKeys = require('../util/JsonPaymentProtocolKeys.js');
+const Bitcore = require('bitcore-lib');
```
Comment (justinkook) [inline packages/bitcore-wallet-client/src/lib/payproV2.ts]: Use CWC.BitcoreLib

File: packages/bitcore-wallet-client/src/lib/api.ts [hunk 3]
```diff
@@ -1522,6 +1526,24 @@ export class API extends EventEmitter {
...
+  getPayProV2(txp) {
+    if (!txp.payProUrl || this.doNotVerifyPayPro) return Promise.resolve();
+
+    const chain = txp.coin ? txp.coin.toUpperCase() : 'BTC';
+
+    return PayProV2.selectPaymentOption(
+      {
+        paymentUrl: txp.payProUrl,
+        chain,
+        currency: '' // TODO
```
Comment (justinkook) [inline packages/bitcore-wallet-client/src/lib/api.ts]: Use `txp.coin.toUpperCase()` for currency. For ERC20 tokens, the `txp.coin` will be the token symbol, and chain will be `'ETH'`

File: packages/bitcore-wallet-client/src/lib/api.ts [hunk 4]
```diff
@@ -1732,52 +1755,58 @@ export class API extends EventEmitter {
...
-            bp_partner_version: this.bp_partner_version,
-
-            // for testing
-            request: this.request
-          },
-          (err, ack, memo) => {
-            if (err) {
-              return cb(err);
+        const chain = txp.coin ? txp.coin.toUpperCase() : 'BTC';
+        const currency = ''; // TODO
```
Comment (justinkook) [inline packages/bitcore-wallet-client/src/lib/api.ts]: Use txp.coin.toUppercase() for currency.

**Comment by cmgustavo** [comment]
I'd like to keep the paths as before: `src/lib/index.ts`

File: packages/bitcore-wallet-client/src/index.ts [hunk 5]
```diff
@@ -7,5 +7,5 @@
 // * Client API.
 // * @alias module:Client.API
 // */
-import { API } from './api';
+import { API } from './lib/api';
```
Comment (cmgustavo) [inline packages/bitcore-wallet-client/src/index.ts]: Why it 's needed?

File: packages/bitcore-wallet-client/package.json [hunk 6]
```diff
@@ -41,13 +41,15 @@
...
     "@types/node": "^10.0.2",
+    "kbpgp": "^2.1.4",
     "bitcore-wallet-service": "^8.8.4",
     "browserify": "^13.1.0",
     "chai": "^1.9.1",
     "coveralls": "^3.0.2",
     "istanbul": "*",
     "mocha": "^5.2.0",
     "mongodb": "^2.0.27",
+    "request-promise": "^4.2.4",
```
Comment (justinkook) [inline packages/bitcore-wallet-client/package.json]: Is this used? I don't see it being used anywhere.

**Pull request merged**